kasa/discover.py

```diff
@@ -171,10 +171,15 @@ class Discover:
     device object.
 
     :func:`discover_single()` can be used to initialize a single device given its
-    IP address. If the type of the device and its IP address is already known,
-    you can initialize the corresponding device class directly without this.
+    IP address. If the :class:`DeviceConfig` of the device is already known,
+    you can initialize the corresponding device class directly without discovery.
 
-    The protocol uses UDP broadcast datagrams on port 9999 for discovery.
+    The protocol uses UDP broadcast datagrams on port 9999 and 20002 for discovery.
+    Legacy devices support discovery on port 9999 and newer devices on 20002.
+
+    Newer devices that respond on port 20002 will most likely require TP-Link cloud
```

Comment:
```suggestion
    Newer devices that respond on port 20002 will require TP-Link cloud
```

Reply:
Devices that have never connected to the cloud should work without authentication so I've updated this to explain that,